src/Services/SocialUserLoginService.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Blumilk\Meetup\Core\Services;
+
+use Blumilk\Meetup\Core\Models\User;
+use Illuminate\Auth\AuthManager;
+use Illuminate\Contracts\Hashing\Hasher;
+use Illuminate\Session\Store;
+
+class SocialUserLoginService
+{
+    public function __construct(
+        public AuthManager $authManager,
+        public Hasher $hasher,
+        public Store $session,
+    ) {}
+
+    public function registerOrLogin($socialUser, string $provider): void
+    {
+        $appUser = User::query()->firstOrCreate(
```

Comment:
Why it's a `$appUser`? Why not `$user`?

Reply:
when it was at the previous location it was easier to tell, but now it's not necessary so i turn it to `$user`